gb-joypad/src/joypad.rs

```diff
@@ -181,20 +181,38 @@ impl Joypad {
     fn update_state(&mut self, scancode: &Option<Scancode>, state: bool) {
         if let Some(scancode) = scancode {
             if let Some(input_type) = self.input_map.get(scancode) {
+                #[cfg(feature = "debug_state")]
+                let mut changed = false;
+                #[cfg(feature = "toggle_joypad")]
+                if state {
+                    self.input_states.insert(
+                        *input_type,
+                        !self.input_states.get(input_type).unwrap_or(&false),
+                    );
+                    #[cfg(feature = "debug_state")]
+                    {
+                        changed = true;
+                    }
```

Comment:
Are those curly bracket necessary ?

Reply:
`error[E0658]: attributes on expressions are experimental` not today ;)